File: pinyougou_manager/pinyougou_manager_interface/src/main/java/com/pinyougou/manager/service/BrandService.java
package com.pinyougou.manager.service;

import com.pinyougou.common.PageResult;
import com.pinyougou.pojo.TbBrand;

import java.util.List;
import java.util.Map;

public interface BrandService {

    //查询品牌列表
    List<TbBrand> findAll();

    //分页查询品牌列表
    public PageResult findPage(Integer pageNum,Integer pageSize);

    //添加品牌
    void add(TbBrand tbBrand);

    //根据id查询品牌
    TbBrand findOne(Long id);

    //修改品牌
    void update(TbBrand tbBrand);

    //删除品牌
    void delete(Long[] ids);

    //条件查询
    PageResult search(Integer pageNum, Integer pageSize, TbBrand tbBrand);

    //查询品牌集合
    List<Map> findBrandList();
}

File: pinyougou_manager_web/src/main/webapp/js/service/brandService.js
app.service('brandService',function ($http) {

    this.findAll = function () {
        return $http.get('../brand/findAll');
    }

    this.findPage=function(pageNum,pageSize) {
        return $http.get('../brand/findPage?pageNum='+pageNum+'&pageSize='+pageSize);
    }

    this.findOne = function (id) {
        return $http.get('../brand/findOne?id='+id);
    }

    this.dele=function (ids) {
        return $http.get('../brand/dele?ids='+ids);
    }

    this.search=function(pageNum,pageSize,searchEntity) {
        return $http.post('../brand/search?pageNum='+pageNum+'&pageSize='+pageSize,searchEntity);
    }

    this.update=function(entity){
        return $http.post('../brand/update',entity);
    }

    this.add=function(entity){
        return $http.post('../brand/add',entity);
    }
    
    this.findBrandList=function () {
        return $http.get('../brand/findBrandList');
    }

});